test/http2/server.go

```diff
@@ -0,0 +1,55 @@
+package http2
```

Comment:
Should we also move the canary server to `test/http/serve-healthcheck.go` to math these 2 new files?
IMO splitting between `cmd/` and `test/` is a little confusing.

Reply:
Maybe we could also combine all 3 of the cobra command functions into one file in `cmd/` called something along the lines of `serve-test-commands.go`.

Reply:
Done in https://github.com/openshift/cluster-ingress-operator/pull/585/commits/489535824d7cf3e70a5c641e48012e589542a725